spl/src/shmem.rs

```diff
@@ -24,13 +24,11 @@ pub fn ret<'a, 'b, 'c, 'info>(
     data: Vec<u8>,
 ) -> ProgramResult {
     let instruction = Instruction {
-        program_id: *ctx.program.key,
+        program_id: solana_program::pubkey::Pubkey::new_unique(), // TODO[paulx]: replace with correct pubkey, should this be the pubkey of the shmem program?
```

Comment:
@armaniferrante this is the only place that explicitly relies on the program key being inside the `CpiContext`. If I'm not mistaken, the `program_id` will always be the id of the `shmem` program so it can just be `id()`?

(I used `Pubkey::new_unique` just as a temporary fix to make it compile)